### DISCUSSION TITLE
Use Fastify as a Connect style middleware

### DISCUSSION THREAD

**Discussion by thefrana** [fastify/fastify#3362] [Q&A] [answered]
Hi,
I may have a silly question, but I am setting up environment with frontend (based on Vite bundler) as well as backend (Fastify, of course). Vite itself supports Connect style middlewares as plugins. I wonder if one can use Fastify as this middleware. Dummy implementation would be to extract url, method and so on from Connect request and pass it to the Fastify with `.inject`, but I think there should be a better way to it. Any idea?

This would also allow Fastify to be used for frontend plugins, which I think is pretty cool expansion.

Thanks

**Comment by galvez** [comment, marked as answer]
@meowalker You may find what you're looking for here: https://fastify-vite.dev — although it goes the other way around — it uses [middie](https://github.com/fastify/middie) to run Vite's development server from your Fastify applications. I'd strongly advise to be [Fastify-first](https://fastify-vite.dev/meta/philosophy.html) is this kind of setup.

**Reply by thefrana** [reply]
Thank you. I was looking at it a month ago, but now it looks like it has documentation. That's great news. I'll give it a try!
